### ISSUE TITLE
Have dependencies files

### ISSUE THREAD

**Issue by lilyminium** [MDAnalysis/mdanalysis#2769]
## Is your feature request related to a problem? ##

It's easier to update one `requirements.txt` and `environment.yml` file than to update all of `.travis.yml`, `.appveyor.yml`, and `azure-pipelines.yml` (potentially), and the user guide, every time we add a new package.

## Describe the solution you'd like ##

Have a `requirements.txt` and `environment.yml` file that we point both CI and people to.

## Describe alternatives you've considered ##

Status quo

**Comment by tylerjereddy** [comment]
I think we should be a bit cautious here on the `pip` side as I mentioned in the cognate PR. Different CI services can behave a little different and the abstraction vs. flexibility to just directly edit the `yml` file is a bit of a tradeoff.

Perhaps more importantly, for `pip` I think we should adopt PEP 518 at this point if we're really going to insist on using files to abstract the deps, etc.

**Comment by lilyminium** [comment]
> Perhaps more importantly, for pip I think we should adopt PEP 518 at this point if we're really going to insist on using files to abstract the deps, etc.

Sure, `pyproject.toml` sounds good to me; `requirements.txt` files were mainly added because I assume people are more used to the `pip install -r requirements.txt` flow.

**Comment by tylerjereddy** [comment]
A bit off-topic but has the dev team rejected the idea of providing wheels for MDA on the package index? That's kind of tied in to this stuff in some ways. It may just be that this would exceed our bandwidth as it does substantially increase the burden of making releases and MDA can be built from source extremely rapidly anyway.

**Comment by tylerjereddy** [comment]
One idea might be to build the wheel in our CI and then install the wheel using `pip` instead of just calling `setup.py`. That way we can still leverage/test `pypyroject.toml` for setting the minimal reqs for building in isolation.

Using some kind of external file to specify what is needed at runtime may still be useful as you say though. Indeed, there are some cool things like `dependabot` then can help maintain those lists in text files automatically.

**Comment by orbeckst** [comment]
I don't think that we rejected wheels, we just don't want to do it ourselves... if there was a service similar to conda-forge then I am sure we'd happily use it. Overall, automating releases is one of the things we'd like to do.

In PR #2808 I added a new Travis job to check that `sdist` produces a file that passes `twine check`. Having more checks that our package actually works would be good as it decreases the manual testing burden at release time.

**Comment by tylerjereddy** [comment]
Right, yeah I have to do the wheels for SciPy and it is a fair bit of work with a separate repo and CI.
